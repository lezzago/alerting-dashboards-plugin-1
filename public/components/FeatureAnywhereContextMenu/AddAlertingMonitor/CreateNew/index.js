/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import CreateNew from './CreateNew.tsx';

export default CreateNew;
