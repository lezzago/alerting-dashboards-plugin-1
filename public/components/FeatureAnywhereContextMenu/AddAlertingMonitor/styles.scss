.add-alerting-monitor {
  height: 100%;
  display: flex;
  flex-direction: column;

  .euiFlyoutBody__overflowContent {
    height: 100%;
    padding-bottom: 0;
  }

  .euiFlexItem.add-alerting-monitor__aside {
    flex: 0 0 auto;
    width: 400px;
    overflow-y: auto;

    &--full {
      width: 100%;
      flex: 1 1 auto;
    }
  }

  &__flex-group {
    height: 100%;
  }

  &__vis {
    height: 400px;

    &--hidden {
      display: none;
    }
  }

  &__flyout {
    &.euiFlyout--small {
      width: calc(400px + 72px);
    }
  }
}
