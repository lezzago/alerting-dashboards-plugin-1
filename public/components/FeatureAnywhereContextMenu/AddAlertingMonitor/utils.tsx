/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { createAugmentVisSavedObject, VisLayerTypes, VisLayerExpressionFn, ISavedAugmentVis, ISavedPluginResource } from '../../../../../../src/plugins/vis_augmenter/public';

export const createSavedObjectAssociation = async (monitorId, visId) => {

  // public static mapping: Record<string, string> = {
  //   title: 'text',
  //   description: 'text',
  //   originPlugin: 'text',
  //   pluginResource: 'object',
  //   visLayerExpressionFn: 'object',
  //   visId: 'keyword,',
  //   version: 'integer',
  // };

  // export interface ISavedPluginResource {
  //   type: string;
  //   id: string;
  // }
  //
  // export interface ISavedAugmentVis {
  //   id?: string;
  //   title: string;
  //   description?: string;
  //   originPlugin: string;
  //   pluginResource: ISavedPluginResource;
  //   visName?: string;
  //   visId?: string;
  //   visLayerExpressionFn: VisLayerExpressionFn;
  //   version?: number;
  // }

  // let pluginResource;
  // pluginResource = ISavedPluginResource('alerting monitor', monitorId);
  const pluginResource: ISavedPluginResource = {
    type: 'alerting monitor',
    id: monitorId,
  };

  // const fn = {
  //   type: VisLayerTypes.PointInTimeEvents,
  //   name: 'overlay_anomalies',
  //   args: {
  //     detectorId: detectorId,
  //   },
  // } as VisLayerExpressionFn;
  //
  // const savedObjectToCreate = {
  //   title: 'test-title',
  //   pluginResourceId: detectorId,
  //   visId: embeddable.vis.id,
  //   visLayerExpressionFn: fn,
  // } as ISavedAugmentVis;

  // enum VisLayerTypes {
  //   PointInTimeEvents = 'PointInTimeEvents',
  // };
  // create the fields needed for the saved obj
  const fn: VisLayerExpressionFn = {
    type: 'PointInTimeEvents',
    name: 'overlay_alerts',
    args: {
      monitorId: monitorId,
    },
  };
  const savedObjectToCreate: ISavedAugmentVis = {
    // id: `${monitorId}-association`,
    title: 'Vis name title',
    description: 'Association to Alerting monitor',
    originPlugin: 'alertingDashboards',
    pluginResource: pluginResource,
    visName: 'Vis name title',
    visId: visId,
    visLayerExpressionFn: fn,
  };
  // export interface ISavedAugmentVis {
  //   id?: string;
  //   title: string;
  //   description?: string;
  //   originPlugin: string;
  //   pluginResource: ISavedPluginResource;
  //   visName?: string;
  //   visId?: string;
  //   visLayerExpressionFn: VisLayerExpressionFn;
  //   version?: number;
  // }

  //console.log('saved obj to create: ', savedObjectToCreate);

  // helper fn to create the saved object given an object implementing the
  // ISavedFeatureAnywhere interface.
  // Note that we actually don't have a hard dep on the feature anywhere loader yet,
  // since we have a dependency on visualizations which has a dependency on the
  // feature anywhere loader. But we will probably need later when
  // using the loader's search functionalities within the UI components.

  // TODO: handle failures if it fails to create
  let newSavedObj = await createAugmentVisSavedObject(savedObjectToCreate);

  // calling save() on the newly-created saved object to actually save it to the system index
  const response = await newSavedObj.save({});

  return response;
};
