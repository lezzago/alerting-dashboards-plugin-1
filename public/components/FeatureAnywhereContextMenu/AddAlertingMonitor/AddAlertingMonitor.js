/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import {
  EuiFlexItem,
  EuiFlexGroup,
  EuiFlyoutHeader,
  EuiFlyoutBody,
  EuiFlyoutFooter,
  EuiTitle,
  EuiSpacer,
  EuiButton,
  EuiButtonEmpty,
  EuiFormFieldset,
  EuiCheckableCard,
} from '@elastic/eui';
import './styles.scss';
import CreateNew from './CreateNew';
import AssociateExisting from './AssociateExisting';

function AddAlertingMonitor({
  embeddable,
  closeFlyout,
  core,
  services,
  flyoutMode,
  setFlyoutMode,
  monitors,
  selectedMonitorId,
  setSelectedMonitorId,
  index,
}) {
  const onCreate = () => {
    const event = new Event('createMonitor');
    document.dispatchEvent(event);
    closeFlyout();
  };

  return (
    <div className="add-alerting-monitor">
      <EuiFlyoutHeader hasBorder>
        <EuiTitle>
          <h2 id="add-alerting-monitor__title">
            {flyoutMode === 'adMonitor' ? 'Set up alerts' : 'Add alerting monitor'}
          </h2>
        </EuiTitle>
      </EuiFlyoutHeader>
      <EuiFlyoutBody>
        <div className="add-alerting-monitor__scroll">
          {flyoutMode !== 'adMonitor' && (
            <>
              <EuiFormFieldset
                legend={{
                  display: 'hidden',
                  children: (
                    <EuiTitle>
                      <span>Options to create a new monitor or associate an existing monitor</span>
                    </EuiTitle>
                  ),
                }}
                className="add-alerting-monitor__modes"
              >
                {[
                  {
                    id: 'add-alerting-monitor__create',
                    label: 'Create new monitor',
                    value: 'create',
                  },
                  {
                    id: 'add-alerting-monitor__existing',
                    label: 'Associate existing monitor',
                    value: 'existing',
                  },
                ].map((option) => (
                  <EuiCheckableCard
                    {...{
                      ...option,
                      key: option.id,
                      name: option.id,
                      checked: option.value === flyoutMode,
                      onChange: () => setFlyoutMode(option.value),
                    }}
                  />
                ))}
              </EuiFormFieldset>
              <EuiSpacer size="m" />
            </>
          )}
          {['create', 'adMonitor'].includes(flyoutMode) && (
            <CreateNew {...{ embeddable, closeFlyout, core, services, index, flyoutMode }} />
          )}
          {flyoutMode === 'existing' && (
            <AssociateExisting
              {...{
                embeddable,
                closeFlyout,
                core,
                services,
                monitors,
                selectedMonitorId,
                setSelectedMonitorId,
              }}
            />
          )}
          <EuiSpacer size="l" />
        </div>
      </EuiFlyoutBody>
      <EuiFlyoutFooter>
        <EuiFlexGroup justifyContent="spaceBetween">
          <EuiFlexItem grow={false}>
            <EuiButtonEmpty onClick={closeFlyout}>Cancel</EuiButtonEmpty>
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiButton onClick={onCreate} fill>
              {flyoutMode === 'existing' ? 'Associate' : 'Create'} monitor
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiFlyoutFooter>
    </div>
  );
}

export default AddAlertingMonitor;
