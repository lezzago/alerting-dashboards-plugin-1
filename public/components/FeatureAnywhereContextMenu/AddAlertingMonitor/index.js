/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import AddAlertingMonitor from './AddAlertingMonitor.tsx';

export default AddAlertingMonitor;
