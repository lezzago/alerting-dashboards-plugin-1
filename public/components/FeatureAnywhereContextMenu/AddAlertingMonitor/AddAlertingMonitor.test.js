/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { shallow } from 'enzyme';
import AddAlertingMonitor from './AddAlertingMonitor';

describe('AddAlertingMonitor', () => {
  test('renders', () => {
    const wrapper = shallow(<AddAlertingMonitor {...{ embeddable: { getTitle: () => '' } }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
