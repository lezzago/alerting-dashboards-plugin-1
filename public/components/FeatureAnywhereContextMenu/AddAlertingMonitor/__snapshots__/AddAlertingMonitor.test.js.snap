// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAlertingMonitor renders 1`] = `
<div
  className="add-alerting-monitor"
>
  <EuiFlyoutHeader
    hasBorder={true}
  >
    <EuiTitle>
      <h2
        id="add-alerting-monitor__title"
      >
        Add alerting monitor
      </h2>
    </EuiTitle>
  </EuiFlyoutHeader>
  <EuiFlyoutBody>
    <div
      className="add-alerting-monitor__scroll"
    >
      <EuiFormFieldset
        className="add-alerting-monitor__modes"
        legend={
          Object {
            "children": <EuiTitle>
              <span>
                Options to create a new monitor or associate an existing monitor
              </span>
            </EuiTitle>,
            "display": "hidden",
          }
        }
      >
        <EuiCheckableCard
          checked={false}
          id="add-alerting-monitor__create"
          key="add-alerting-monitor__create"
          label="Create new monitor"
          name="add-alerting-monitor__create"
          onChange={[Function]}
          value="create"
        />
        <EuiCheckableCard
          checked={false}
          id="add-alerting-monitor__existing"
          key="add-alerting-monitor__existing"
          label="Associate existing monitor"
          name="add-alerting-monitor__existing"
          onChange={[Function]}
          value="existing"
        />
      </EuiFormFieldset>
      <EuiSpacer
        size="m"
      />
      <EuiSpacer
        size="l"
      />
    </div>
  </EuiFlyoutBody>
  <EuiFlyoutFooter>
    <EuiFlexGroup
      justifyContent="spaceBetween"
    >
      <EuiFlexItem
        grow={false}
      >
        <EuiButtonEmpty>
          Cancel
        </EuiButtonEmpty>
      </EuiFlexItem>
      <EuiFlexItem
        grow={false}
      >
        <EuiButton
          fill={true}
          onClick={[Function]}
        >
          Create
           monitor
        </EuiButton>
      </EuiFlexItem>
    </EuiFlexGroup>
  </EuiFlyoutFooter>
</div>
`;
